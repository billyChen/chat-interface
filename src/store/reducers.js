import { combineReducers } from "redux";
import MessageReducer from './message/reducers'



export default combineReducers(
    {
        messages: MessageReducer
    }
)   