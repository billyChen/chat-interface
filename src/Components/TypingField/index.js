import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import clsx from 'clsx'

// Material-ui
import { makeStyles } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import OutlinedInput from '@material-ui/core/OutlinedInput';
import InputAdornment from '@material-ui/core/InputAdornment';
import InputLabel from '@material-ui/core/InputLabel'
import FormControl from '@material-ui/core/FormControl'
import IconButton from '@material-ui/core/IconButton'
import Button from '@material-ui/core/Button'
import VisibilityIcon from '@material-ui/icons/Visibility';
import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';

// Actions
import { addMessage } from '../../store/message/reducers'


const useStyles = makeStyles({
    root: {

    },
    inputText: {
        width: '80%',
        marginTop: '15px'
    },
    button: {
        height: '50%',
        marginLeft: '15px',
        fontWeight: '600'
    },
    isDisabled: {
        color: '#cad1d9',
        pointerEvents: 'none',
        cursor: 'not-allowed',
        backgroundColor: '#f4f6f7',
    }
})

const TypingField = () => {

    const classes = useStyles()
    const dispatch = useDispatch()

    const [message, setMessage] = useState('')
    const [isPublic, setIsPublic] = useState('public')

    const handleTextChange = (e) => {
        setMessage(e.target.value)
    }

    const handleSendMessage = () => {
        dispatch(addMessage({ message, isPublic }))
        setMessage('')
    }

    const handleSetVisibility = () => {
        setIsPublic(!isPublic)
    }

    return (
        <Box display="flex" alignItems="baseline">
            <FormControl fullWidth className={clsx(classes.margin, classes.textField)} variant="outlined">
                <InputLabel htmlFor="outlined-adornment-message">Votre message</InputLabel>
                <OutlinedInput
                    data-testid="input-message"
                    id="outlined-adornment-message"
                    onChange={handleTextChange}
                    className={classes.inputText}
                    label="Search field"
                    type="search"
                    variant="outlined"
                    value={message}
                    endAdornment={
                        <InputAdornment position="end">
                            <IconButton onClick={handleSetVisibility} edge="end">
                                {isPublic ? <VisibilityIcon /> : <VisibilityOffIcon />}
                            </IconButton>
                        </InputAdornment>
                    }
                />
            </FormControl>
            <Button id="submit-button" onClick={handleSendMessage} className={clsx(classes.button, {
                [classes.isDisabled]: message.length === 0
            })} variant="contained">Envoyer</Button>
        </Box >
    )
}

export default TypingField