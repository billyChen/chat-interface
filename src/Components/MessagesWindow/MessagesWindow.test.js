
import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'
import { Provider } from 'react-redux'
import TypingField from '.'
import { render, fireEvent, screen } from '../../store/test-utils'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

const mockStore = configureStore([]);

describe('MessagesWindow Component', () => {
    let component;
    let store;
    beforeEach(() => {
        store = mockStore({
            messages: [{
                message: 'abc',
                isPublic: true,
                date: Date.now()
            }]
        },
        );
        store.dispatch = jest.fn();
        component = renderer.create(
            <Provider store={store}>
                <TypingField />
            </Provider>
        );
    })

    it('should display the component', () => {
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('should display several messages', () => {
        expect(component.toJSON().children.filter((c) => c.type === 'div')[0].children.length).toBe(1)
    })

})