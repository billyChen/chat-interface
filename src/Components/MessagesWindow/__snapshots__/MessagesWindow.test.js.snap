// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesWindow Component should display the component 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorTransparent MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        Messagerie
      </h5>
    </div>
  </header>
  <div
    className="MuiBox-root MuiBox-root-2 makeStyles-root-1"
  >
    <div
      className="MuiBox-root MuiBox-root-5 makeStyles-root-3"
    >
      abc
      <div
        className="MuiBox-root MuiBox-root-6 makeStyles-messageInfos-4"
      >
        Public
         - 
        02/02/2021
      </div>
    </div>
  </div>
</div>
`;
